charts/shoot-core/components/charts/kube-proxy/templates/kube-proxy-cleanup-script.yaml

```diff
@@ -0,0 +1,30 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-proxy-cleanup-script
+  namespace: kube-system
+  labels:
+    app: kubernetes
+    gardener.cloud/role: system-component
+    origin: gardener
+    role: proxy
+data:
+  cleanup.sh: |
+    #!/bin/sh -e
+    OLD_KUBE_PROXY_MODE="$(cat "$1")"
+    if [ -z "${OLD_KUBE_PROXY_MODE}" ] || [ "${OLD_KUBE_PROXY_MODE}" == "${KUBE_PROXY_MODE}" ]; then
+      echo "${KUBE_PROXY_MODE}" >"$1"
+      exit 0
+    fi
+    {{- if semverCompare "< 1.17" .Values.kubernetesVersion }}
+    /hyperkube
+    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
+    proxy
+    {{- else }}
+    kube-proxy
+    {{- end }}
+    {{- else }}
+    /usr/local/bin/kube-proxy
+    {{- end }} --v=2 --cleanup --proxy-mode="${OLD_KUBE_PROXY_MODE}"
```

Comment:
```
W0513 06:07:39.225390       8 server.go:213] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
```

According to this message the `--proxy-mode` flag should not be needed, correct?

Reply:
Yes, but unfortunately it is not working without the `--proxy-mode` flag. Everytime it returns nil for the ipvs interface, in the  CleanupLeftovers func. https://github.com/kubernetes/kubernetes/blob/77a86a52c0ec88bed49e752831c7820a98ca1c69/pkg/proxy/ipvs/proxier.go#L776.